assets/wizards/setup/views/welcome/index.js

```diff
@@ -1,50 +1,198 @@
+import { NewspackLogo, Button, ProgressBar, withWizardScreen } from '../../../../components/src';
 
-/**
- * Location Setup Screen.
- */
-class Welcome extends Component {
-	componentDidMount() {
+const POST_COUNT = newspack_aux_data.is_e2e ? 12 : 40;
+
+const REQUIRED_SOFTWARE_SLUGS = [
+	'jetpack',
+	'amp',
+	'pwa',
+	'wordpress-seo',
+	'google-site-kit',
+	'newspack-blocks',
+	'newspack-theme',
+];
+
+const TOTAL = POST_COUNT + 4 + REQUIRED_SOFTWARE_SLUGS.length;
+
+const ERROR_TYPES = {
+	plugin_configuration: { message: __( 'Plugin installation' ) },
+	starter_content: { message: __( 'Starter content' ) },
+};
+
+const starterContentFetch = endpoint =>
+	apiFetch( {
+		path: `/newspack/v1/wizard/newspack-setup-wizard/starter-content/${ endpoint }`,
+		method: 'post',
+	} );
+
+const Welcome = ( { buttonAction } ) => {
+	const [ installationProgress, setInstallationProgress ] = useState( 0 );
+	const [ softwareInfo, setSoftwareInfo ] = useState( [] );
+	const [ errors, setErrors ] = useState( [] );
+	const addError = errorInfo => error =>
+		setErrors( _errors => [ ..._errors, { ...errorInfo, error } ] );
+
+	useEffect( () => {
 		document.body.classList.add( 'newspack_page_newspack-setup-wizard__welcome' );
-	}
-
-	componentWillUnmount() {
-		document.body.classList.remove( 'newspack_page_newspack-setup-wizard__welcome' );
-	}
-
-	/**
-	 * Render.
-	 */
-	render() {
-		return (
-			<>
-				<div className="newspack-logo__wrapper welcome">
-					<NewspackLogo centered />
-				</div>
-				<div className="newspack-setup-wizard__welcome">
-					<h2>{ __( 'Welcome to WordPress for your Newsroom!' ) }</h2>
-					<p>
-						{ __(
-							'We’ll help you get set up by installing the most relevant Newspack plugins in the background.'
-						) }
-					</p>
-				</div>
-			</>
+
+		Promise.all(
+			REQUIRED_SOFTWARE_SLUGS.map( item =>
+				apiFetch( { path: `/newspack/v1/plugins/${ item }` } ).then( res =>
+					setSoftwareInfo( _softwareInfo => [ ..._softwareInfo, res ] )
+				)
+			)
+		);
+
+		return () => document.body.classList.remove( 'newspack_page_newspack-setup-wizard__welcome' );
+	}, [] );
+
+	const increment = () => setInstallationProgress( progress => progress + 1 );
+
+	const install = async () => {
+		increment();
+
+		// Plugins and theme.
+		const softwarePromises = softwareInfo.map( item => {
+			if ( item.Status === 'active' ) {
+				increment();
+				return () => Promise.resolve();
+			}
+			return () =>
+				apiFetch( {
+					path: `/newspack/v1/plugins/${ item.Slug }/configure/`,
+					method: 'POST',
+				} )
+					.then( increment )
+					.catch( addError( { info: ERROR_TYPES.plugin_configuration, item: item.Name } ) );
+		} );
+		for ( let i = 0; i < softwarePromises.length; i++ ) {
+			await softwarePromises[ i ]();
+		}
+
+		// Starter content.
+		await starterContentFetch( `categories` )
+			.then( increment )
+			.catch( addError( { info: ERROR_TYPES.starter_content, item: 'categories' } ) );
```

Comment:
The error messages for if these fail are pretty blunt—could we add some short description, e.g. "Failed to create starter content: homepage" or "Failed to install plugin: [PLUGIN NAME]"?

<img width="505" alt="Screen Shot 2021-02-05 at 11 59 56 AM" src="https://user-images.githubusercontent.com/2230142/107077148-b04e7d80-67a9-11eb-8c50-5b0abf1bba05.png">

Reply:
Done in 8267401ad3cc851fed4a29794a5b79a36e75c1a2